### PR TITLE
#402 Update code to get correct command offset.

### LINKED ISSUE #402: //os.args does not handle arrai CLI global flags
Please **do not post any internal, closed source snippets** on this public issue tracker!

### Description
Running `arrai -d r file.arrai` can change the value `//os.args` as it shifts the values of the CLI arguments.

### Steps to Reproduce
1. Create file.arrai that prints `//os.args`
2. Run `arrai -d r file.arrai arg1 arg2 arg3` 

**Expected behavior:**
`['file.arrai', 'arg1', 'arg2', 'arg3']`

**Actual behavior:**
`['r', 'file.arrai', 'arg1', 'arg2', 'arg3']`

### PR DESCRIPTION
**Pull request by ericzhang6222**
Fixes #402 .

Changes proposed in this pull request:
- 
- 

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: syntax/std_os_nonwasm.go [hunk 1]
```diff
@@ -21,12 +22,21 @@ func stdOsGetArgs() rel.Value {
...
 }
 
+func lookupCommandOffset() int {
+	for index, item := range os.Args[1:] {
+		if !strings.HasPrefix(item, "-") {
+			return index + 2
+		}
+	}
+	return 2
+}
```
Comment (marcelocantos) [inline syntax/std_os_nonwasm.go:38]: See if it's cleaner to get the `run` and `eval` commands to supply the arguments directly.

Comment (ericzhang6222) [reply syntax/std_os_nonwasm.go]: Fixed.

File: tools/cmd_util.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		(isatty.IsCygwinTerminal(os.Stdin.Fd()) && isatty.IsCygwinTerminal(os.Stdout.Fd()))
+}
+
+//SetArgs set context arguments, for example command line `arrai -d r file.arrai arg1 arg2 arg3` will
+//save `file.arrai arg1 arg2 arg3` as a slice to Arguments.
+func SetArgs(c *cli.Context) {
+	Arguments = Arguments.With("Arguments", c.Args().Slice())
+}
+
+var Arguments = frozen.NewMap([]frozen.KeyValue{}...)
```
Comment (marcelocantos) [inline tools/cmd_util.go:22]:
```suggestion
var Arguments = []string
```

File: syntax/std_os_nonwasm.go [hunk 3]
```diff
@@ -7,24 +7,15 @@ import (
...
-			return rel.NewArray()
-		} else {
-			offset = 2
-		}
+	args, found := tools.Arguments.Get("Arguments")
+	if found {
+		return strArrToRelArr(args.([]string))
 	}
-	return strArrToRelArr(os.Args[offset:])
+	return rel.NewArray()
```
Comment (marcelocantos) [inline syntax/std_os_nonwasm.go:18]: No need for a special case.

**Pull request merged**